### PR TITLE
feat: support DTS bundle file name

### PR DESCRIPTION
**Pull request by Timeless0911**
## Summary

support generate bundle DTS name according to entry name.

image

## Checklist

- [ ] Tests updated (or not required).
- [ ] Documentation updated (or not required).

### CODE REVIEW

File: packages/plugin-dts/src/utils.ts [hunk 1]
```diff
@@ -70,3 +71,26 @@ export async function processDtsFiles(
...
+      name: Object.keys(entryConfig)[0] as string,
+      path: Object.values(entryConfig)[0] as string,
+    };
+  }
+
+  logger.warn(
+    '@microsoft/api-extractor only support single entry of Record<string, string> type to bundle DTS.',
+  );
+
+  return {
```
Comment (fi3ework) [inline packages/plugin-dts/src/utils.ts:92]: Is dts will be emitted if return from here?

File: packages/plugin-dts/src/index.ts [hunk 2]
```diff
@@ -54,10 +60,19 @@ export const pluginDts = (options: PluginDtsOptions): RsbuildPlugin => ({
           stdio: 'inherit',
         });
 
+        let dtsEntry: DtsEntry = {
+          name: undefined,
+          path: undefined,
+        };
```
Comment (fi3ework) [inline packages/plugin-dts/src/index.ts:66]: `processSourceEntry` could directly return a `dtsEntry`, instead another mut processing.

File: packages/plugin-dts/src/utils.ts (hunk 1 above)
Comment (Timeless0911) [reply packages/plugin-dts/src/utils.ts:92]: The bundless DTS will generated and api-extractor will throw error
image

Comment (fi3ework) [reply packages/plugin-dts/src/utils.ts:92]: 1. bundleless output is useless as user want the bundled, can we detect the invalid option earlier and do not call the bundleless emit.
2. the log above (in line 88) should turn to be an error.

Comment (Timeless0911) [reply packages/plugin-dts/src/utils.ts:92]: I do a judgement and log like this
image

Comment (fi3ework) [reply packages/plugin-dts/src/utils.ts:92]: Could put the warn right above the error?

Comment (Timeless0911) [reply packages/plugin-dts/src/utils.ts:92]: image
is this OK?

**Review by fi3ework — APPROVED** [review]

**Pull request merged**